torchvision/models/swin_transformer.py

```diff
@@ -266,11 +488,20 @@ def __init__(
         attention_dropout: float = 0.0,
         stochastic_depth_prob: float = 0.0,
         norm_layer: Callable[..., nn.Module] = nn.LayerNorm,
-        attn_layer: Callable[..., nn.Module] = ShiftedWindowAttention,
+        attn_layer: Optional[Callable[..., nn.Module]] = None,
+        version: int = 1,
```

Comment:
Undo this change.
```suggestion
        attn_layer: Callable[..., nn.Module] = ShiftedWindowAttention,
```

We shouldn't pass version. Instead we can pass a boolean flag that allows doing pre_normalization (better name can be used here). This flag will be explosed to the model builders and they will be responsible for passing the right `att_layer` and bool value to construct the network for V1 and V2. This will align your code to the idioms we used at Vision.